src/main/java/net/barrage/school/java/ecatalog/app/ProductServiceImpl.java

```diff
@@ -28,8 +36,16 @@ public List<Product> listProducts() {
         return products;
     }
 
+    @Scheduled(fixedDelayString = "${ecatalog.products.fetchDelay}")
+    @CacheEvict(value = "products", allEntries = true)
+    public void clearProductCache() {
+    }
+
     @Override
     public List<Product> searchProducts(String query) {
+        // I read something about cache not working if called from method from same bean, so this will never call
```

Comment:
Yeap! That's great u've debug it)

There is a trick for thsi case - self inject
https://www.baeldung.com/spring-self-injection

Reply:
Thanks. Should I leave it for now or implement self inject?